# generic message type for action specification.
time stamp
string type
string[] fields
string[] values